### PR TITLE
Elasticsearch: Unify default value for geo hash grid precision across the code to 3

### PR DESCRIPTION
**Pull request by ivanahuckova**
To be reviewed after https://github.com/grafana/grafana/pull/73917 is merged.

While working on https://github.com/grafana/grafana/pull/73917, I notice couple of confusing things:
1. In a backend code where we are creating `GeoHashGrid` struct, we are using a default value of `5` `innerAgg := &GeoHashGridAggregation{
		Field:     field,
		Precision: 5,
	}`. This is different from a code where we are setting a default value of precision to `3`  in `a.Precision = bucketAgg.Settings.Get("precision").MustInt(3)`. The code where the default value is `5` is always overwritten by `3`, so it does not matter what value is there. But we need to have there a value, cause the overwriting is not guaranteed (for example if other parts of the code would use it), but we should have the same default value. 

I was also doing some investigating work to find out when this was introduced - and it was this way from the beginning. The confusion is probably that `5` is the default value from `Elasticsearch` when precision is not set. And we decided to use `3` in the UI. But this whole 5, or 3 creates confusion in UI, so I think it make sense to unify it and make `3` default for all cases.

With this change to backend code, nothing should change as `5` is in current implementation always overwritten by `3`. 
	
2. In frontend code we had a logic for description `const precision = Math.max(Math.min(parseInt(bucketAgg.settings?.precision || '5', 10), 12), 1);` that:

- was adding incorrect `5` default value if no value was added (however, we were adding 3 when initialising query editor)
- adjusting description to use values 1-12, but that would mean that even if user typed `300` (which is incorrect), the incorrect vaule would be used and query would error out, but the description would show falesly-correct `12` which would make debugging of issue harder. 

I have decided to unify default value to `3` as that's what frontend behaviour was when precision wasn't touched/filled by users.

Before:

https://github.com/grafana/grafana/assets/30407135/3de79e0b-8fb4-4d2b-940f-f51c2c02d46e

After:

https://github.com/grafana/grafana/assets/30407135/cb43f85e-de2a-47a4-84c7-50f6a52818bd

### CODE REVIEW

File: pkg/tsdb/elasticsearch/client/search_request.go [hunk 1]
```diff
@@ -9,6 +9,7 @@ const (
 	HighlightPreTagsString  = "@HIGHLIGHT@"
 	HighlightPostTagsString = "@/HIGHLIGHT@"
 	HighlightFragmentSize   = 2147483647
+	DefaultPrecision        = 3
```
Comment (svennergr) [inline pkg/tsdb/elasticsearch/client/search_request.go:12]: Just to be slightly more precisive, what do you think about adding "geoHash" to the name?

```suggestion
	DefaultGeoHashPrecision        = 3
```

**Review by svennergr — APPROVED** [review]
Just one nit, the rest LGTM!

File: pkg/tsdb/elasticsearch/client/search_request.go (hunk 1 above)
Comment (ivanahuckova) [reply pkg/tsdb/elasticsearch/client/search_request.go]: Fixed in https://github.com/grafana/grafana/pull/73922/commits/fffe7471ead9d952a14796d6558680e591c4b989

File: public/app/features/dashboard-scene/serialization/DashboardsLoader.ts
[the remark this answers is not in the document]
Comment (ivanahuckova) [reply public/app/features/dashboard-scene/serialization/DashboardsLoader.ts:1]: Yes, looking at it now.

**Pull request merged**